streamflow-api.yaml

```diff
@@ -0,0 +1,768 @@
+      properties:
+        times:
+          description: Time coordinates, specified as ISO8601 date-time strings.
+          type: array
+          items:
+            type: string
+          example: ["2000-01-15", "2000-02-15"]
+        values:
+          type: array
+          items:
+            type: number
+          example: [5000, 6000]
+      required:
+        - times
+        - values
+
+    ### Hydromodel outputs
+
+    # Individual hydromodel outputs
+
+    HydromodelOutputId:
+      description: Unique id of hydromodel output that drives a streamflow computation
+      type: integer
+      example: 4
+
+    HydromodelOutputState:
+      description: State for a hydromodel output. Contains complete state.
+      allOf:
+        - type: object
+          properties:
+            hydromodel_output_id:
+              $ref: '#/components/schemas/HydromodelOutputId'
+            model:
+              type: string
+              example: CanESM2
+            experiment:
+              type: string
+              example: rcp85
+            etc:
+              type: string
+              example: ... more tbd ...
+
+    HydromodelOutputHypermedia:
+      description: Hypermedia for a thing
+      type: object
+      properties:
+        links:
+          allOf:
+            - $ref: '#/components/schemas/HypermediaLinkList'
+            - example:
+                - rel: 'self'
+                  uri: 'http://apis.paccifcclimate.org/hydromodel_output/4'
+
+    HydromodelOutput:
+      description: Full representation of a hydromodel output - state + hypermedia.
+      allOf:
+        - $ref: '#/components/schemas/HydromodelOutputState'
+        - $ref: '#/components/schemas/HydromodelOutputHypermedia'
+
+    # Lists (collections) of hydromodel outputs
+
+    HydromodelOutputListItem:
+      description: Representation of a single item in a list of hydromodel outputs.
+      type: object
+      allOf:
+        - type: object
+          properties:
+            uri:
+              type: string
+              description: URI of hydromodel output
+              example: 'http://apis.paccifcclimate.org/hydromodel_output/4'
+        - $ref: '#/components/schemas/HypermediaLinkList'
+        - $ref: '#/components/schemas/HydromodelOutputState'
+
+    HydromodelOutputList:
+      description: List of hydromodel outputs in collection.
+      type: array
+      items:
+        $ref: '#/components/schemas/HydromodelOutputListItem'
+      example:
+        - uri: 'http://apis.paccifcclimate.org/hydromodel_output/3'
+          model: "CanESM2"
+          experiment: "rcp45"
+        - uri: 'http://apis.paccifcclimate.org/hydromodel_output/4'
+          model: "CanESM2"
+          experiment: "rcp85"
+
+    ### Streamflow orders
+
+    # Individual streamflow orders
+
+    StreamflowOrderStateNew:
+      description: Defines a new order for a streamflow result set.
+      type: object
+      properties:
+        hydromodel_output_id:
```

Comment:
I _think_, but am not sure, that the way to indicate the hydromodel output desired as input to this computation is with such an id. This has the virtue of simplicity but it exposes an internal implementation detail, and that is regarded as undesirable in resource-oriented architectures (RESTful architecture).

The alternative would be the hyperlink to the hydromodel output. This does not expose an internal, but it does require the backend to translate the hyperlink to an actual hydromodel output record, which is more roundout.

I'm not sure which to choose.